third_party/move/move-compiler-v2/src/pipeline/control_flow_graph_simplifier.rs

```diff
@@ -0,0 +1,599 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Prerequisites: none.
+//!
+//! Simplifies the control flow graph in the following ways:
+//! (Implemented by `ControlFlowGraphSimplifier`)
+//! 1. Eliminates branch or jump to jump:
+//!     Any empty block
+//!
+//!     L1: goto L2 where L1 != L2
+//!
+//!     is removed, and any block jumping to L1 will jump to L2 directly
+//!
+//!     goto L1
+//!     => (the => here means the instruction above will be replaced by the instruction below)
+//!     goto L2
+//!
+//!     if ... goto L1 else ...
+//!     =>
+//!     if ... goto L2 else ...
+//!
+//!     if ... goto ... else L1
+//!     =>
+//!     if ... goto ... else L2
+//!
+//! 2. Removes edges where the source has only one successor and the target has only one predecessor:
+//!     If there is block
+//!     BB1: xx
+//!          goto L2
+//!     and block
+//!
+//!     BB2: L2
+//!          yy
+//!
+//!     which has only one predecessor BB1.
+//!     Then we move BB2 to the end of BB1 so that BB1 becomes
+//!
+//!     BB1: xx
+//!          yy
+//!
+//!     and BB2 is removed.
+//!
+//! 3. Removes all jumps to the next instruction:
+//!     Whenever we have
+//!
+//!     goto L;
+//!     L;
+//!
+//!     replace it by simply
+//!
+//!     L;
```

Comment:
In addition, if there is no reference to a label `L` (e.g., because of this transformation), then the label `L` can also be removed?

Reply:
We don't need to do that; labels are anyway not in the final stackful bytecode.

Reply:
@fEst1ck 
Here is why we should do this.

In the current file format generator, when we see a `Label`, it can cause additional stack flushes, see https://github.com/aptos-labs/aptos-core/blob/main/third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs#L213

Even otherwise, cleaning up stackless bytecode, while you are already at it, is a good idea.

Reply:
Done